### PR TITLE
[Enhancement] Search Entire Commit Message on Demand; fixes #337

### LINKED ISSUE #337: [Feature Request] comment-changes: Check comments both body and summary for keywords
It would be useful to give the users the flexibility to use both options 'Squash and merge' and  the normal merge PR in their repos and still have all of them documented properly with Aeruginous by creating a logic to make it intelligently choose the right source to analyze from.

The logic I have in mind is to search for the categories in both comment's  summery and  body and choose the one that contains one of the categories. And if not it will fallback to body if the -b flag is active or to summery otherwise.

For that we can add a new argument like `--intelligent_search` which can be combined with `--body` and `--fallback_category`

### PR DESCRIPTION
**Pull request by kevinmatthes**
This PR fixes #337.

### CODE REVIEW

File: src/changelog/comment_changes.rs [hunk 1]
```diff
@@ -50,6 +50,10 @@ pub struct CommentChanges {
   #[arg(long, short = 'C')]
   fallback_category: Option<String>,
 
+  /// Whether to enforce the fragment creation.
+  #[arg(long, short = 'F')]
+  force: bool,
```
Comment (kevinmatthes) [inline src/changelog/comment_changes.rs:55]: The option to trigger the search for changes in the opposite message part is named `--force`, abbreviated by `-F`.

File: src/changelog/comment_changes.rs [hunk 2]
```diff
@@ -296,15 +303,28 @@ impl Logic {
...
+                    } else if self.cli.force {
+                      if let Some((category, change)) =
+                        self.harvest_message(if self.cli.body {
+                          summary.trim()
+                        } else {
+                          body.trim()
+                        })
+                      {
+                        Self::insert(&mut result, category, change);
+                      }
```
Comment (kevinmatthes) [inline src/changelog/comment_changes.rs:327]: If the first search in the preferred message part did not succeed, the opposite message part will be searched *if* the force mode is active.  The results of the second search will be applied in any case:

- both queries fail → no fragment
- second query succeeds → fragment from second query
- second query succeeds due to fallback category → fragment from second query

**Review by kevinmatthes — COMMENTED** [review]
@AmmarAbouZor

Just for completeness, I commented the most important aspects of my solution.  In general, it matches your feature request exactly -- except for the renamed trigger option.  I will release these changes within the next two days.

**Comment by AmmarAbouZor** [comment]
@kevinmatthes 
This looks great very thanks for implementing it :1st_place_medal:

**Pull request merged**